metabase/metabase
# Migrate cross-version e2e tests using a published SDK package from release branch

Part of https://github.com/metabase/metabase/issues/50113

### Description

This adds a new CI workflow to run SDK component tests against the stable SDK build for release branch.

This is migrated from https://github.com/metabase/metabase/pull/49196 , but also adds logic to resolve sdk version and checkout to this version tag in git to get a stable set of e2e tests that existed at the moment of sdk release.
This should help to avoid failures for functionality that was merged, but haven't been released to the npm yet.

e.g. for release x.51 -
```
yarn add @metabase/embedding-sdk-react@0.51.9
```

> [!NOTE]
> backport most likely will fail and we will have to resolve conflicts 

### How to verify

Check CI results - https://github.com/metabase/metabase/actions/runs/12033324150/job/33547730386?pr=50423
This is done with hardcoded branch name - `release-x.51.x`

Run tests locally from this branch - `yarn test-cypress-open-component-sdk` (you will also need EE env vars for Cypress)

### Checklist
